### PR TITLE
fix: flip attention marker to avoid nested-emphasis fusion

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and discussions and couldn’t find anything or linked relevant results below 
* [x] I made sure the docs are up to date
* [x] I included tests (or that’s not needed)

### Description of changes

Problem: emphasis uses the configured marker unconditionally, so
`emphasis > emphasis > text` serializes as `**a**`, which re-parses
as `strong > text`. The same path erases the outer type from
`strong > emphasis > text` (`***a***` re-parses as `emphasis > strong`)
and collapses any strict three-or-more nested emphasis chain
(`***a***` re-parses as `emphasis > strong` rather than three
nested emphases).

https://github.com/syntax-tree/mdast-util-to-markdown/issues/12 catalogued a broader family of related shapes:
`***emphasis*in emphasis*`, `*a*_b__`, `a ***b*c d*`, and variants
with different leading run lengths. As wooorm noted in the thread,
"pulling a thread somewhere will have something happen somewhere
entirely different": marker choice in one place interacts with
flanking rules elsewhere, and covering every shape needs
coordination beyond picking a single marker per node.

Scope: land the minimal change that fixes the shapes guaranteed to
drift under the current serializer, without regressing shapes where
CommonMark's attention algorithm already recovers the original tree
through fusion. Escape-based work on the remaining https://github.com/syntax-tree/mdast-util-to-markdown/issues/12 shapes is
left for follow-up.

Approach: introduce `lib/util/emphasis-marker.js`. Both the
emphasis handler and its peek route marker selection through it,
keeping lookahead in `container-phrasing` consistent with what the
handler emits. The helper flips in two narrow situations:

1.  The emphasis is the only child of an attention parent (emphasis
    or strong), and both its opening and closing markers would sit
    immediately next to the parent's primary marker. Using the
    opposite marker (`*_a_*`, `**_a_**`) breaks the fusion into
    strong or em+strong.

2.  The emphasis sits at the top of a strict same-type chain of
    depth two or more (every link has exactly one emphasis child)
    when the primary marker is `*`. Three-deep emphasis only
    round-trips with `_` on the outside, because `_`'s flanking
    rules are stricter than `*`'s. The rule is asymmetric on
    purpose: with primary `_` the first rule's adjacency flip
    alternates correctly on its own.

Strong is never flipped. A run of four asterisks already pairs as
two strong delimiters, six as three, and so on, so strong
round-trips without help.

Journey (what was tried and why the scope narrowed):

- An earlier iteration flipped strong too and regressed ~18 corpus
  fixtures whose nested-strong shapes relied on long fused runs of
  asterisks. Strong was dropped from the flip to recover them.

- Flipping whenever `info.before` or `info.after` matches the
  primary caused cascading flips on paragraph-level attention
  siblings: `[emphasis, strong, emphasis]` serialized as
  `_a___a__*a*`, where `_` + `__` at the em/strong boundary
  re-tokenised as a single `___` run. The flip was narrowed to
  attention parents only.

- Widening rule 1 to first-or-last-child of any attention parent
  fixed several https://github.com/syntax-tree/mdast-util-to-markdown/issues/12 shapes (`***emphasis*in emphasis*`,
  `***x*y z*`, `****x*y z*`) but regressed `***a*a*-*`
  (`emphasis > [emphasis > [emphasis, text], text]`): CommonMark's
  rule 17 uses the leading `***` fusion to recover the three-deep
  structure, and the flip broke that recovery. The rule was
  tightened to only-child plus strict-chain.

- The only-child formulation plus the strict-chain rule is the
  widest version verified to cause zero transitions from ok to
  finding across 600 corpus files (commonmark, gfm, all
  configurations).

Edge cases covered by new tests:

- Plain emphasis and strong, with primaries `*` and `_`, showing
  the helper is inert on non-nested attention.
- `emphasis > emphasis` with each of primary `*` and `_`, yielding
  `*_a_*` and `_*a*_`.
- `strong > emphasis` yielding `**_a_**`.
- `emphasis > strong` and `strong > strong` preserved at `***a***`
  and `****a****`, proving strong is untouched.
- Strict three-deep emphasis chains with both primaries, both
  yielding `_*_a_*_` (chain flip vs adjacency flip arrive at the
  same output by different routes).
- Emphasis parents with more than one child, demonstrating the
  only-child guard preserves shapes vanilla handles.
- Middle-sibling emphasis, confirming no flip at non-boundary
  positions.
- Top-level `[emphasis, strong, emphasis]` round-trip preserved.
- `***a*a*-*` fusion shape explicitly preserved as a regression
  guard against future widening.
- Round-trips for parsed `*_a_*`, parsed `_*_a_*_`, synthesised
  `emphasis > emphasis`, synthesised `strong > emphasis`, and a
  three-deep chain preceded by a text sibling.

Refs: https://github.com/syntax-tree/mdast-util-to-markdown/issues/12

Disclosure: written with AI assistance

### CODE REVIEW

File: lib/util/emphasis-marker.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  ) {
+    return other
+  }
+
+  if (primary === '*' && strictChainDepth(node) >= 2) return other
+
+  return primary
+}
+
+/**
```
[the remark this answers is not in the document]
Comment (ChristianMurphy) [reply lib/util/emphasis-marker.js:63]: observation is correct: `strictChainDepth(node) >= 2` does match multiple nodes in a strict chain of depth 4 or deeper, which I'll grant wasn't by explicit design. But applying the proposed gating is a regression, not an improvement. Comparison across depths:

| depth | current output | current reparses as | copilot output | copilot reparses as |
|---|---|---|---|---|
| 2 | `*_a_*` | `em > em > text("a")` ✅ | `*_a_*` | `em > em > text("a")` ✅ |
| 3 | `_*_a_*_` | `em > em > em > text("a")` ✅ | `_*_a_*_` | same ✅ |
| 4 | `__*_a_*__` | `strong > em > em > text("a")` | `_*_*a*_*_` | `[em(text("*")), em("a"), em(text("*"))]` |
| 5 | `___*_a_*___` | `em > strong > em > em > text("a")` | `_*_*_a_*_*_` | 4 flat `em` with literal `*` text |
| 6 | `____*_a_*____` | `strong > strong > em > em > text("a")` | `_*_*_*a*_*_*_` | 4 flat `em` with literal `*` text |

Both approaches drift at depth 4+, CommonMark has no round-tripping representation for a strict 4-deep emphasis chain. But the current output is strictly more faithful to the input tree:

- The text content (`"a"`) is preserved.
- Attention nesting is preserved (the inner `em > em` survives).
- Only the outermost element type changes (em → strong).

The proposed fix:
- Flattens the nesting entirely into sibling `em` nodes.
- Injects literal `*` or `_` characters into text nodes that weren't in the original tree (content corruption).

So the `__...__` at the outer boundary looks alarming, but it's producing the least-bad degradation at depth 4+. I'd rather keep the accidental over-match and document it than trade it for a drift that corrupts content.

**Review by Murderlon — APPROVED** [review]
Looks high-quality 👍 . Double checked with Devin review as well and no issues found: https://app.devin.ai/review/syntax-tree/mdast-util-to-markdown/pull/74
